### ISSUE TITLE
environment > configuration > source activate is ignored

### ISSUE THREAD

**Issue by sacdallago** [debbiemarkslab/EVcouplings#152]
In `/n/groups/marks/users/assaf/dor/BB22OPb` check the `Immunity_config.txt` and `Immunity.failed` files. There is a major inconsistency:

In the config:
```
environment:
  configuration:
  - source activate evcouplings_develop
```

but in the stack trace:
```
Traceback (most recent call last):
  File "/n/groups/marks/software/anaconda_o2/envs/evcouplings_stable/lib/python3.6/site-packages/pandas/core/indexes/base.py", line 2525, in get_loc
    return self._engine.get_loc(key)
  File "pandas/_libs/index.pyx", line 117, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 139, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 1265, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 1273, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'alignment_id'
...
```

Notice, that `evcouplings_stable` env is being used instead of the one indicated in the config. This needs fixing before _submission_, otherwise, we cannot claim reproducibility.

The reason why this phenomenon occurred in this case, is that the source being activated upon submission via CLI was `evcouplings_stable`, a.k.a.: it seems that the env being considered is the one from the login node submitting the job, which is very odd.

**Comment by b-schubert** [comment]
The worker run as child processes and call the command via ```subprocesses.Popen```, with ```shell = True```

**Comment by thomashopf** [comment]
Okay great so the environment solution should also work there... which means the above proposed solution is 100% consistent across different engines

**Comment by aggreen** [comment]
I want to share a weird environment behavior I encountered on o2 today that may be related to this problem or may be independent. It probably needs to be considered for the solution. 

If I use a particular environment, eg evcouplings_develop, to submit a job via the command line app, the run will fail if the configuration is set to "source activate evcouplings_develop." This behavior is new as of this weekend (as far as I can tell). The reason is that each conda environment has its own conda installation which becomes the default path, so when the submitted job tries to "source activate evcouplings_develop," the environment is not found because it belongs to the parent environment, and we get the following error. 

```
CondaEnvironmentNotFoundError: Could not find environment: evcouplings_develop .
You can list all discoverable environments with `conda info --envs`.
```

Bobby encountered this error earlier this week so I know that it's not a problem specific to me.

**Comment by thomashopf** [comment]
If you first `source activate evcouplings_develop` and then sbatch the command "which python; which evcouplings", you will get 
```
/n/groups/marks/software/anaconda_o2/envs/evcouplings_develop/bin/python
/n/groups/marks/software/anaconda_o2/envs/evcouplings_develop/bin/evcouplings
```
I guess this means that unlike on the old Orchestra, on o2 the submitted job inherits the environment from the submitting session. As evcouplings_develop is already active, the CondaEnvironmentNotFoundError occurs.

(This is assuming there is no source activate evcouplings_develop in any of .bashrc, .bash_profile or .profile - at which point putting the source activate into the configuration line doesn't make sense anyways, and you would expect exactly the same error).

Bottom line: To me this looks like new behaviour of o2, and the right action for that environment is not to use `source activate evcouplings_dev` when submitting jobs on o2. Doesn't look to me like an evcouplings issue (whereas the point further up definitely is one).

**Comment by thomashopf** [comment]
Fix for the actual issue in first post is implemented and will come shortly.

**Comment by aggreen** [comment]
Hey Thomas, thanks for clarification on my issue. I'll update
evcouplings_monomer_o2 config and the email to our group to reflect this
new environment behavior.

**Comment by thomashopf** [comment]
Thanks Anna for updating. 

The initial issue should also be gone now since the config runner and the summarizer are now being called by their applications `evcouplings_runcfg` and `evcouplings_summarize`, so what ever these link to in the environment where the job is executed will be used.

That being said, I would strongly encourage anyone to always submit from the same environment that is being used for the actual computation. With the (apparent) inheritance of the parent environment on o2 through sbatch and removing the source activate from our default config file, there should be less room for accidentally using a wrong environment.

(PS: develop environment not yet updated on o2 - I won't touch this).

**Comment by sacdallago** [comment]
Thanks both.

I think 
[this one](https://github.com/debbiemarkslab/EVcouplings/issues/152#issuecomment-377356790) can be easily fixed with puppet from RC's side (have sessions include anaconda in their `$PATH`). I'll put it on my todo, but let's first solve more urgent matters with them...

I'll update `evcouplings_develop` on o2 on Monday after making sure with Anna I'm up to speed with all current developments
